File: tests/integration/construct_component_plain/python/component.py
# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

from typing import Any, Optional

import pulumi

class Component(pulumi.ComponentResource):
    def __init__(self,
                 name: str,
                 opts: Optional[pulumi.ResourceOptions] = None,
                 children: Optional[int] = None):
        props = dict()
        props["children"] = children
        super().__init__("testcomponent:index:Component", name, props, opts, True)

File: tests/integration/construct_component_output_values/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

export interface BarArgs {
    tags?: pulumi.Input<{[key: string]: pulumi.Input<string>}>;
}

export interface FooArgs {
    something?: pulumi.Input<string>;
}

export interface ComponentArgs {
    bar?: pulumi.Input<BarArgs>;
    foo?: FooArgs;
}

export class Component extends pulumi.ComponentResource {
    constructor(name: string, args?: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, args, opts, true /*remote*/);
    }
}
